src/fsharp/DotNetFrameworkDependencies.fs

```diff
@@ -34,114 +34,78 @@ module internal FSharp.Compiler.DotNetFrameworkDependencies
-        let pattern = "\"name\": \"" + tfmPrefix
-        let startPos =
-            let startPos = file.IndexOf(pattern, StringComparison.OrdinalIgnoreCase)
-            if startPos >= 0  then startPos + (pattern.Length) else startPos
-
-        let length =
-            if startPos >= 0 then
-                let ep = file.IndexOf("\"", startPos)
-                if ep >= 0 then ep - startPos else ep
-            else -1
-        match startPos, length with
-        | -1, _
-        | _, -1 -> None
-        | pos, length -> Some ("netcoreapp" + file.Substring(pos, length))
-
-
-    let getFrameworkRefsPackDirectoryPath =
-        match executionTfm with
-        | Some _ ->
-            let appRefDir = Path.Combine(getFSharpCompilerLocation, "../../../packs/Microsoft.NETCore.App.Ref")
-            if Directory.Exists(appRefDir) then
-                Some appRefDir
+    //     if that changes we will need to find another way to do this.  Hopefully the sdk will eventually provide an API
+    //     use the well know location for obj to traverse the file system towards the
+    //
+    //          packs\Microsoft.NETCore.App.Ref\sdk-version\netcoreappn.n
+    //     we will rely on the sdk-version match on the two paths to ensure that we get the product that ships with the
+    //     version of the runtime we are executing on
+    //     Use the reference assemblies for the highest netcoreapp tfm that we find in that location.
+    let version, frameworkRefsPackDirectoryRoot =
+        try
+            let version = DirectoryInfo(implementationAssemblyDir).Name
+            let microsoftNETCoreAppRef = Path.Combine(implementationAssemblyDir, "../../../packs/Microsoft.NETCore.App.Ref")
+            if Directory.Exists(microsoftNETCoreAppRef) then
+                Some version, Some microsoftNETCoreAppRef
             else
-                None
-        | _ -> None
+               Some version,  None
+        with | _ -> None, None
 
     let isInReferenceAssemblyPackDirectory filename =
-        match getFrameworkRefsPackDirectoryPath with
-        | Some appRefDir ->
+        match frameworkRefsPackDirectoryRoot with
+        | Some root ->
             let path = Path.GetDirectoryName(filename)
-            path.StartsWith(appRefDir, StringComparison.OrdinalIgnoreCase)
+            path.StartsWith(root, StringComparison.OrdinalIgnoreCase)
         | _ -> false
 
-    let getFrameworkRefsPackDirectory =
-        match executionTfm, getFrameworkRefsPackDirectoryPath with
-        | Some tfm, Some appRefDir ->
+    let frameworkRefsPackDirectory =
+        let tfmPrefix = "netcoreapp"
+        let tfmCompare c1 c2 =
+            let deconstructTfmApp (netcoreApp: DirectoryInfo) =
+                let name = netcoreApp.Name
+                try
+                    if name.StartsWith(tfmPrefix, StringComparison.InvariantCultureIgnoreCase) then
+                        Some (Double.Parse(name.Substring(tfmPrefix.Length)))
```

Comment:
This will fail in non-English scenarios where the decimal separator is a comma.  It should be a simple string split: `name.Substring(tfmPrefix.Length).Split(".", 2)`.